opennms-webapp-rest/src/main/java/org/opennms/web/rest/v1/ResourceRestService.java

```diff
@@ -123,4 +140,21 @@ public ResourceDTO getResourceForNode(@PathParam("nodeCriteria") final String no
 
         return ResourceDTO.fromResource(resource, depth);
     }
+
+
+    @POST
+    @Path("generateId")
+    @Consumes({MediaType.APPLICATION_JSON})
+    @Produces({MediaType.APPLICATION_JSON})
+    public Response saveResourcesWithId(String[] resources) {
+
+        String resourcesInJson = m_gson.toJson(resources);
```

Comment:
Should we verify that all the given resource IDs are valid at insertion time?

Reply:
I want the `POST` call to be responsive enough. That's the reason I don't even check `key` to exist.